File: leetcode/338. Counting Bits/CountingBits.go
func countBits(n int) []int {
	res := make([]int, 0)
	i := 0
	for i <= n {
		num := numberOnes(i)
		res = append(res, num)
		i++
	}
	return res
}

func numberOnes(n int) int {
	res := 0
	for n > 0 {
		temp := n % 2
		res += temp
		n = n >> 1
	}
	return res
}

File: leetcode/338. Counting Bits/CountingBits.cpp
class Solution
{
public:
    vector<int> countBits(int n)
    {
        vector<int> res;
        for (int i = 0; i <= n; i++)
        {
            int bits = numOneBits(i);
            res.emplace_back(bits);
        }
        return res;
    }

    int numOneBits(int n)
    {
        int count = 0;
        while (n > 0)
        {
            int temp = n % 2;
            count += temp;
            n >>= 1;
        }
        return count;
    }
};